src/lib.rs

```diff
@@ -185,6 +209,9 @@ pub async fn start_server(
                 .layer(Extension(Arc::new(service_config.clone())))
                 .layer(Extension(Arc::new(network_config)))
                 .layer(Extension(Arc::new(Mutex::new(DispenseTracker::new(clock)))))
+                .layer(Extension(Arc::new(
+                    Box::new(auth_handler) as Box<dyn AuthHandler>
```

Comment:
Without this cast requests will return with an internal server error without further context. Took me a while to figure this one out